src/app/share/services/redirections.service.ts

```diff
@@ -0,0 +1,65 @@
+import { t } from 'i18next';
+import navigationService from '../../core/services/navigation.service';
+import { AppView } from '../../core/types';
+import notificationsService, { ToastType } from '../../notifications/services/notifications.service';
+import shareService from './share.service';
+import { History } from 'history';
+import { AdvancedSharedItem } from '../types';
+
+/**
+ * Handles access to a private shared folder, fetching its content and navigating accordingly.
+ *
+ * @param {Object} options - Options object containing parameters.
+ * @param {string} options.folderUUID - The UUID of the folder to access.
+ * @param {History} options.history - The history object for navigation.
+ * @param {(sharedFolder: AdvancedSharedItem) => void} options.navigateToFolder - Function to navigate to the accessed folder.
+ * @param {() => void} options.onError - Function to be executed if an error occurs during the process.
+ */
+const handlePrivateSharedFolderAccess = ({
```

Comment:
I think this function is hiding a lot of logic, maybe splitting this into multiple pieces will help to maintain + understand it.

`getPrivateSharedFolderAccess` +  `redirectBySharedFolderAccess` maybe?